mongolog/handlers.py

```diff
@@ -231,14 +154,40 @@ def check_keys(self, record):
             return record
 
         for k, v in record['msg'].items():
-            record['msg'][self.new_key(k)] = record['msg'].pop(k)
-
-            if isinstance(v, dict):
-                for old_key in record['msg'][k].keys():
-                    record['msg'][k][self.new_key(old_key)] = record['msg'][k].pop(old_key)
+            self._check_keys(k, v, record['msg'])
 
         return record
 
+    def _check_keys(self, k, v, _dict):
+        """
+        Recursively check key's for special characters.
+        """
+        _dict[self.new_key(k)] = _dict.pop(k)
+
+        if isinstance(v, dict):
+            for nk, vk in v.items():
+                self._check_keys(nk, vk, v)
```

Comment:
Does this need another branch to check if it's a list containing objects with further keys?

Reply:
Yep you are right.  Just did some quick testing and it fails with that currently.  I'll update and push changes a bit later.